DBZ-4597 Add Oracle FAQ entry about "Logs don't contain offset" (#3159)
debezium/debezium: documentation/modules/ROOT/pages/connectors/oracle.adoc

The rest of this section describes how {prodname} handles various kinds of faults and problems.

[id="oracle-logs-do-not-contain-offset-perform-new-snapshot"]
=== Logs do not contain offset, perform a new snapshot

When the {prodname} Oracle connector restarts, you may find that the connector reports the following error:

[source]
----

> Online REDO LOG files or archive logs do not contain the offset scn xxxxxxx. Please perform a new snapshot.

Reviewer: For the MySQL connector we have the [`when_needed`](https://debezium.io/documentation/reference/stable/connectors/mysql.html#mysql-property-snapshot-mode) snapshot mode; looks like that'd be a worthwhile addition for the Oracle connector, too? If you agree, @Naros, can you log an issue for this in Jira?

Author: Done, logged [DBZ-4612](https://issues.redhat.com/browse/DBZ-4612).

----

This error occurs when the connector examines the available redo and archive logs and was unable to locate a record that contains the SCN in the connector offsets.
Since the connector automatically starts where it left off and if that SCN (position) is no longer available, then a re-snapshot will be necessary.

It's worth noting that you may find a record in the `V$ARCHIVED_LOG` table that has a SCN range that matches the referenced SCN; however, that alone doesn't mean the record is available to be mined.
The record must also have a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.
If any of these criteria do not match, then the record is considered incomplete and cannot be mined.

Archive logs must be retained up-to if not slightly longer than your longest downtime window of the connector.

[NOTE]
====
When a record has no value in the `NAME` column, this means the log has been removed the file system.

> In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to represent the log has been deleted.

Reviewer 2: ```suggestion
In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to represent the log has been deleted.
```
In such records, the  value of the `DELETED` field is set to `YES`, and the `STATUS` field is set to `D`  to indicate that the log is deleted.

Reviewer 2: In the preceding, is it the _log_ that is deleted or the _record_? That is, should the sentence be:

In this situation, the record is accompanied with `DELETED` being set to `YES` and the `STATUS` column being set to `D` to indicate the record was removed.

Author: No, _log_ is correct in this context.  We basically have a record in the database that previously pointed to the log; however, when the log gets removed, the record is kept but the references to the log on the filesystem are nulled out, i.e. `NAME` is set to `NULL`, `DELETED` is set to `YES`, and `STATUS` is set to `D` to indicate deletion.

The rest of this section describes how {prodname} handles various kinds of faults and problems.

[id="oracle-logs-do-not-contain-offset-perform-new-snapshot"]
=== Logs do not contain offset, perform a new snapshot

> When the {prodname} Oracle connector restarts, you may find that the connector reports the following error:

Reviewer 2: ```suggestion
In some cases, after the {prodname} Oracle connector restarts, it reports the following error:
```

[source]
----
Online REDO LOG files or archive logs do not contain the offset scn xxxxxxx. Please perform a new snapshot.
----

After the connector examines the redo and archive logs, if it cannot find the SCN that is recorded in the connector offsets, it returns the preceding error.
Because the connector uses the SCN to determine where to resume processing, if the expected SCN if not found, a new snapshot must be completed.

You might find that the `V$ARCHIVED_LOG` table contains a record with an SCN that matches the expected range.
However, the record might not be available for mining.

> The record must also have a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.

Reviewer 2: ```suggestion
To be available for mining, a record must include a filename in the `NAME` column, a value of `NO` in the `DELETED` column, and a value of `A` (available) in the `STATUS` column.
```

> If any of these criteria do not match, then the record is considered incomplete and cannot be mined.

Reviewer 2: ```suggestion
If a record does not match any of these criteria, it is considered incomplete and cannot be mined.
```
